apiserver/src/webhook/mod.rs

```diff
@@ -0,0 +1,141 @@
+mod request;
+mod response;
+
+pub use self::request::{ConversionRequest, Request};
+pub use self::response::{ConversionResponse, ConvertResult, Response};
+
+use lazy_static::lazy_static;
+use models::node::Convert;
+use snafu::{ResultExt, Snafu};
+use std::collections::{HashMap, HashSet};
+use tracing::{event, Level};
+
+type ConvertMethod = fn(serde_json::Value) -> models::node::Result<serde_json::Value>;
+pub type Result<T> = std::result::Result<T, WebhookConvertError>;
+
+lazy_static! {
+    static ref CONVERTERS: HashMap<String, (String, ConvertMethod)> = {
+        let mut converters = HashMap::new();
+        converters.insert(
+            "brupop.bottlerocket.aws/v1".to_string(),
+            (
+                "brupop.bottlerocket.aws/v2".to_string(),
+                models::node::v2::BottlerocketShadow::convert_from_old_version as ConvertMethod,
+            ),
+        );
+        converters
+    };
+    static ref TARGET_SET: HashSet<String> = {
+        let mut target_set = HashSet::new();
+        for value in CONVERTERS.values() {
+            target_set.insert(value.0.to_string());
+        }
+        target_set
+    };
+}
+
+pub fn convert_request_to_response(req: &ConversionRequest) -> ConversionResponse {
```

Comment:
I would like hear some suggestions in this file to better struct the methods and some advice on the unit test.

Reply:
Updated the code based on the comments